test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -364,22 +361,16 @@ protected final TestFeatureService createTestFeatureService(
         Map<String, Set<String>> clusterStateFeatures,
         Set<Version> semanticNodeVersions
     ) {
-        // Historical features information is unavailable when using legacy test plugins
-        boolean hasHistoricalFeaturesInformation = System.getProperty("tests.features.metadata.path") != null;
-
         final List<FeatureSpecification> featureSpecifications = new ArrayList<>(createAdditionalFeatureSpecifications());
         featureSpecifications.add(new RestTestLegacyFeatures());
```

Comment:
@ldematte Do you know if there's a reason why `RestTestLegacyFeatures` aren't provided via `createAdditionalFeatureSpecifications` as done for `YamlTestLegacyFeatures`?

Reply:
Very simple: RestTestLegacyFeatures predates createAdditionalFeatureSpecifications.
If you feel like it, you can refactor it (but it's very minor IMO)